### DISCUSSION TITLE
How I can debug cfltk code

### DISCUSSION THREAD

**Discussion by lbk861** [fltk-rs/fltk-rs#1509] [Q&A] [answered]
I'm using VSCode with codelldb to debug a fltk app.When I want to step into the code like `Fl_xx`, debugger always steps over it.
Also I try using lldb attaching the program in term, which can be found C++ functions name, but no source code found.
Is there one way to debug this?

**Comment by MoAlyousef** [comment, marked as answer]
fltk-rs builds FLTK in Release mode, this is the default because on windows not doing so would complicate the build (different search dirs depending on the cmake build generator, and different lib names like fltkd.lib etc for msvc only). 

So to debug FLTK with fltk-rs, you would need to clone fltk-rs, change the cmake Release flag here:
https://github.com/fltk-rs/fltk-rs/blob/6633faae385812971d460dfc195b4c44b9d2c211/fltk-sys/build/source.rs#L182

You can point your project to your clone in your Cargo.toml using the path to the clone.

**Reply by lbk861** [reply]
Yes. Something strange is that for my app, it doesn't work.But I create one new app, it works.
